Shoko.CLI/Worker.cs

```diff
@@ -1,69 +1,101 @@
+#region
 using System;
+using System.ComponentModel;
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.Extensions.Hosting;
 using NLog;
-using Shoko.Server;
+using Shoko.Server.Commands;
 using Shoko.Server.Server;
 using Shoko.Server.Settings;
 using Shoko.Server.Utilities;
+#endregion
+namespace Shoko.CLI;
 
-namespace Shoko.CLI
+public class Worker : BackgroundService
 {
-    public class Worker : BackgroundService
+    private static readonly Logger                    Logger = LogManager.GetCurrentClassLogger();
+    private readonly        IHostApplicationLifetime? _appLifetime;
+
+    public Worker() { }
+    public Worker(IHostApplicationLifetime lifetime) => _appLifetime = lifetime;
+
+    protected override Task ExecuteAsync(CancellationToken stoppingToken)
     {
-        private static readonly Logger Logger = LogManager.GetCurrentClassLogger();
+        SetInstanceIfNeeded();
+        InitInstanceLogger();
+        LoadSettings();
+        if (ServerCouldNotStart())
+            return Task.CompletedTask;
+        EnsureAniDBSocketInitialized();
+        AddEventHandlers();
+        return Task.CompletedTask;
+    }
 
-        public ProgramArguments Args { get; set; }
-        private IHostApplicationLifetime AppLifetime { get; set; }
-        
-        public Worker()
-        {
-            Args = null;
-        }
+    /**
+     * Ensure that the AniDB socket is initialized. Try to Login, then start the server if successful.
+     */
+    private static void EnsureAniDBSocketInitialized()
+    {
+        if (ServerSettings.Instance.FirstRun is false)
+            ShokoServer.RunWorkSetupDB();
+        else
+            Logger.Warn("The Server is NOT STARTED. It needs to be configured via webui or the settings.json");
+    }
 
-        public Worker(ProgramArguments args, IHostApplicationLifetime lifetime)
-        {
-            Args = args;
-            AppLifetime = lifetime;
-        }
+    private static bool ServerCouldNotStart() => ShokoServer.Instance.StartUpServer() is false;
 
-        protected override async Task ExecuteAsync(CancellationToken stoppingToken)
-        {
-            if (!string.IsNullOrEmpty(Args?.Instance)) ServerSettings.DefaultInstance = Args.Instance;
+    private static void InitInstanceLogger() => ShokoServer.Instance.InitLogger();
+    
+    private static string? GetInstanceFromCommandLineArguments()
+    {
+        const int notFound = -1;
+        var       args     = Environment.GetCommandLineArgs();
```

Comment:
I personally don't care, but our current formatter settings would nuke that whitespace. I'm less anal about that stuff, as long as it makes sense for readability in the current context.